qa/restart-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/NeuralSparseSearchIT.java

```diff
@@ -0,0 +1,113 @@
+        waitForClusterHealthGreen(NODES_BWC_CLUSTER);
+        if (isRunningAgainstOldCluster()) {
+            String modelId = uploadSparseEncodingModel();
+            loadModel(modelId);
+            createPipelineProcessor(modelId, PIPELINE_NAME);
+            createIndexWithConfiguration(
+                getIndexNameForTest(),
+                Files.readString(Path.of(classLoader.getResource("processor/SparseIndexMappings.json").toURI())),
+                PIPELINE_NAME
+            );
+
+            addSparseEncodingDoc(
+                getIndexNameForTest(),
+                "0",
+                List.of(TEST_SPARSE_ENCODING_FIELD),
+                List.of(testRankFeaturesDoc1),
+                List.of(TEST_TEXT_FIELD),
+                List.of(TEXT_1)
+            );
+        } else {
+            Map<String, Object> pipeline = getIngestionPipeline(PIPELINE_NAME);
+            assertNotNull(pipeline);
+            String modelId = TestUtils.getModelId(pipeline, SPARSE_ENCODING_PROCESSOR);
+            loadModel(modelId);
+            addSparseEncodingDoc(
+                getIndexNameForTest(),
+                "1",
+                List.of(TEST_SPARSE_ENCODING_FIELD),
+                List.of(testRankFeaturesDoc2),
+                List.of(TEST_TEXT_FIELD),
+                List.of(TEXT_2)
+            );
+            validateTestIndex(modelId);
+            deletePipeline(PIPELINE_NAME);
+            deleteModel(modelId);
+            deleteIndex(getIndexNameForTest());
+        }
+
+    }
+
+    private void validateTestIndex(String modelId) throws Exception {
+        int docCount = getDocCount(getIndexNameForTest());
+        assertEquals(2, docCount);
+        BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+        NeuralSparseQueryBuilder sparseEncodingQueryBuilder = new NeuralSparseQueryBuilder().fieldName(TEST_SPARSE_ENCODING_FIELD)
+            .queryText(TEXT_1)
+            .modelId(modelId);
+        MatchQueryBuilder matchQueryBuilder = new MatchQueryBuilder(TEST_TEXT_FIELD, TEXT_1);
+        boolQueryBuilder.should(sparseEncodingQueryBuilder).should(matchQueryBuilder);
+        Map<String, Object> response = search(getIndexNameForTest(), boolQueryBuilder, 1);
+        Map<String, Object> firstInnerHit = getFirstInnerHit(response);
+
+        assertEquals("0", firstInnerHit.get("_id"));
+        float minExpectedScore = computeExpectedScore(modelId, testRankFeaturesDoc1, TEXT_1);
+        assertTrue(minExpectedScore < objectToFloat(firstInnerHit.get("_score")));
+    }
+
+    private String uploadSparseEncodingModel() throws Exception {
+        String requestBody = Files.readString(
+            Path.of(classLoader.getResource("processor/UploadSparseEncodingModelRequestBody.json").toURI())
+        );
+        return registerModelGroupAndGetModelId(requestBody);
+    }
+
+    private String registerModelGroupAndGetModelId(String requestBody) throws Exception {
```

Comment:
I see lots of duplication of methods including this one as well. Can we avoid code duplication?